### PR TITLE
Bugfix/scalable linking

### PR DESCRIPTION
**Pull request by geekbeast**
In Progress

### CODE REVIEW

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java [hunk 1]
```diff
@@ -164,21 +181,29 @@ public void revoke(
...
-
-            return result;
-        } catch ( SQLException e ) {
-            logger.error( "Unable to load all entity sets", e );
-            return ImmutableList.of();
-        }
+    public PostgresIterable<EntitySet> getAllEntitySets() {
+        return new PostgresIterable<>(
+                () -> {
+                    try {
```
Comment (makosblade) [inline src/main/java/com/openlattice/edm/PostgresEdmManager.java]: Prefer 
```
try ( Connection connection = hds.getConnection();
                        PreparedStatement ps = connection.prepareStatement( getAllEntitySets ) ) {
```

Comment (geekbeast) [reply src/main/java/com/openlattice/edm/PostgresEdmManager.java]: This will break the postgres iterable. I hope you haven't done this anywhere :O

Comment (makosblade) [reply src/main/java/com/openlattice/edm/PostgresEdmManager.java]: 😅

**Comment by geekbeast** [comment]
Ready to go.

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java [hunk 2]
```diff
@@ -59,7 +75,8 @@
  * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
  */
 public class PostgresEdmManager implements DbEdmManager {
-    private static final Logger logger = LoggerFactory.getLogger( PostgresEdmManager.class );
+    private static final String getAllEntitySets = "SELECT * FROM " + PostgresTable.ENTITY_SETS.getName();
```
Comment (makosblade) [inline src/main/java/com/openlattice/edm/PostgresEdmManager.java]: Not a blocking comment but do we plan on eventually having a single file of all of our PG queries somewhere? Might be nice to have a `ReadQueries.getAllEntitySets` vs `WriteQueries.updateEntityMetadata`

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java [hunk 3]
```diff
@@ -164,21 +181,29 @@ public void revoke(
...
-            logger.error( "Unable to load all entity sets", e );
-            return ImmutableList.of();
-        }
+    public PostgresIterable<EntitySet> getAllEntitySets() {
+        return new PostgresIterable<>(
+                () -> {
+                    try {
+                        Connection connection = hds.getConnection();
+                        PreparedStatement ps = connection.prepareStatement( getAllEntitySets );
+                        ps.setFetchSize( 10000 );
```
Comment (makosblade) [inline src/main/java/com/openlattice/edm/PostgresEdmManager.java]: `AutoCommit = false` 😀

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java (hunk 2 above)
Comment (geekbeast) [reply src/main/java/com/openlattice/edm/PostgresEdmManager.java]: Most of them are in PostgresQueries except for specific queries that are contained in *QueryService classes.

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java (hunk 3 above)
Comment (geekbeast) [reply src/main/java/com/openlattice/edm/PostgresEdmManager.java]: Done.

**Pull request merged**